src/bentoml/_internal/cloud/deployment.py

```diff
@@ -1,73 +1,260 @@
+                if target["config"]["runners"] is not None:
+                    for _, config in target["config"]["runners"].items():
+                        if config.get("hpa_conf") is None:
+                            config["hpa_conf"] = hpa_conf_dct
+
+        if resource_instance:
+            for target in dct["targets"]:
+                if target["config"].get("resource_instance") is None:
+                    target["config"]["resource_instance"] = resource_instance
+                if target["config"]["runners"] is not None:
+                    for _, config in target["config"]["runners"].items():
+                        if config.get("resource_instance") is None:
+                            config["resource_instance"] = resource_instance
+
+        create_deployment_schema = bentoml_cattr.structure(dct, CreateDeploymentSchema)
+        logger.info(f"{create_deployment_schema.name} is created.")
+        return cls._create_deployment(
+            context=context,
+            cluster_name=cluster_name,
+            create_deployment_schema=create_deployment_schema,
+        )
+
+    @classmethod
+    def list(
+        cls, context: str | None = None, cluster_name: str | None = None
     ) -> DeploymentListSchema:
 
         yatai_rest_client = get_rest_api_client(context)
-        if clusterName:
-            res = yatai_rest_client.get_deployment_list(clusterName)
+        if cluster_name:
+            res = yatai_rest_client.get_deployment_list(cluster_name)
         else:
             res = yatai_rest_client.get_deployment_list(default_context_name)
         if res is None:
             raise BentoMLException("List deployments request failed")
         return res
 
-    def create_deployment(
-        self,
+    @classmethod
+    def create_from_file(
+        cls,
+        path: str,
         context: str | None = None,
-        clusterName: str | None = None,
-        *,
-        json_content: str,
+        cluster_name: str | None = None,
+        path_context: str | None = None,
     ) -> DeploymentSchema:
+        with open(path, "r"):
+            data = json.load(resolve_user_filepath(path, path_context))
 
         yatai_rest_client = get_rest_api_client(context)
-        if clusterName is None:
-            clusterName = default_context_name
-        deployment_schema = schema_from_json(json_content, CreateDeploymentSchema)
+        if cluster_name is None:
+            cluster_name = default_context_name
+        deployment_schema = bentoml_cattr.structure(data, CreateDeploymentSchema)
         for target in deployment_schema.targets:
             res = yatai_rest_client.get_bento(target.bento_repository, target.bento)
             if res is None:
                 raise BentoMLException(
                     f"Create deployment: {target.bento_repository}:{target.bento} does not exist"
                 )
         res = yatai_rest_client.get_deployment(
```

Comment:
does this mean res in L226 will always be overwritten by res in L231? I guess we don't need the res in L226 anw?